### PR TITLE
Character Animation

### PR DESCRIPTION
**Pull request by Raketfart**
Added a CharacterAnimation class, and hooked it up to Character and CharacterSpriteController.
The helmet is still a seperate sprite, so only the character is animated.
See issue #142

### CODE REVIEW

File: Assets/Scripts/Controllers/CharacterSpriteController.cs [hunk 1]
```diff
@@ -57,10 +57,24 @@ public void OnCharacterCreated(Character c)
         char_go.transform.SetParent(characterParent.transform, true);
 
         SpriteRenderer sr = char_go.AddComponent<SpriteRenderer>();
-        sr.sprite = SpriteManager.current.GetSprite("Character", "p2_front");
+        //sr.sprite = SpriteManager.current.GetSprite("Character", "p2_front");
```
Comment (Grenkin1988) [inline Assets/Scripts/Controllers/CharacterSpriteController.cs]: Should not be commented out code

**Comment by Grenkin1988** [comment]
You should run StyleCop, looks like there are some issues. (For example to many empty lines)

**Comment by Raketfart** [comment]
Commented line is removed, and I ran StyleCop on CharacterAnimation.
There are MANY stylecop issues in Character and CharacterSpriteController.
Should I clean up those too? Or will that mess too much with other pull requests?

**Comment by Fogsight** [comment]
It is probably the best for clean up to be performed separately, it can be snuck through when nobody working on the file, harmlessly.

File: Assets/Scripts/Models/Character.cs [hunk 2]
```diff
@@ -329,7 +338,7 @@ private bool CheckForJobMaterials()
                 // Walk towards a tile containing the required goods.
                 Debug.Log("Walk to the stuff");
                 Debug.Log(myJob.canTakeFromStockpile);
-
+
```
Comment (Mizipzor) [inline Assets/Scripts/Models/Character.cs]: Theres a handful of lines in this file with trailing white space. Could you fix that? :)

**Comment by Raketfart** [comment]
@Mizipzor I removed the whitespace now.

File: Assets/Scripts/Models/Character.cs [hunk 3]
```diff
@@ -138,6 +138,15 @@ public Tile JobTile
...
     public Inventory inventory;
 
+    // holds all character animations
+    public CharacterAnimation animation;
+
+    // is the character walking or idle
+    public bool IsWalking;
+
+    // 0=north, 1=east, 2=south, 3=west
+    public int Facing;
```
Comment (TomMalbran) [inline Assets/Scripts/Models/Character.cs]: Maybe we should use an enum here. It would make things look more clear.

**Comment by Mizipzor** [comment]
This is "just" a refactor, right? The character should be equally animated before and after this is merged?

**Comment by Raketfart** [comment]
@Mizipzor Uhm, what do you mean? The character wasn't animated before
This PR makes his feet move and his facing match his direction.

**Comment by Raketfart** [comment]
This is what it looks like:
animation

**Comment by Tranberry** [comment]
should work with #684

**Comment by Mizipzor** [comment]
I couldve sworn my character was animated before I tested this. I *might* be crazy, let me have another look.

**Comment by Mizipzor** [comment]
Nope, he wasnt. Either I was running with your changes when I thought I wasnt, or I'm actually crazy. 

Good work on the enum though, I would've asked for that as well. :)

Good job!

**Pull request merged**